game-core/src/test/java/games/strategy/engine/data/RouteFinderTest.java

```diff
@@ -0,0 +1,72 @@
+package games.strategy.engine.data;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RouteFinderTest {
```

Comment:
The test coverage here looks pretty good for just a single test.  The only suggestions I'd add are for some boundary conditions:

* When `start` and `end` are the same.
* When no route exists between `start` and `end`.